sympy/core/expr.py

```diff
@@ -3425,6 +3425,14 @@ def compute_leading_term(self, x, logx=None):
         as_leading_term is only allowed for results of .series()
         This is a wrapper to compute a series first.
         """
+        from sympy.utilities.exceptions import SymPyDeprecationWarning
+
+        SymPyDeprecationWarning(
+            feature="compute_leading_term",
+            useinstead="as_leading_term",
+            issue=21843,
+            deprecated_since_version="1.12"
+        ).warn()
```

Comment:
This deprecation warning doesn't work:
```python
In [1]: x.compute_leading_term(x)
...
TypeError: SymPyDeprecationWarning.__init__() got an unexpected keyword argument 'feature'
```
This is why we need tests for things that are deprecated. This has already been released in SymPy 1.12 and 1.13 now and is broken so at this stage it would be best just to delete this method rather than fixing the deprecation warning.